contracts/DemoToken.sol

```diff
@@ -8,7 +8,7 @@ import "openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol";
  * Creator starts with 1 million tokens
  */
 contract DemoToken is ERC20Mintable, ERC20Detailed {
-    constructor(string name, string symbol) ERC20Detailed(name, symbol, 18) public {
+    constructor(string memory name, string memory symbol) ERC20Detailed(name, symbol, 18) public {
```

Comment:
not sure if it matters, but is the string argument calldata or memory?

Reply:
when I try calldata:
/home/heynow/streamr/monoplasma/contracts/DemoToken.sol:11:17: TypeError: Data location must be "memory" for parameter in function, but "calldata" was given.
    constructor(string calldata name, string memory symbol) ERC20Detailed(name, symbol, 18) public {

Reply:
I don't entirely understand the new rules.

External fns require calldata for strings: I guess if you do a function call within a contract, you pass a pointer to memory location but if you call another contract, you must load it into calldata because the other contract can't see memory of the calling contract.

Public fns use memory kw for strings? Maybe it always copies from calldata to memory.

See https://solidity.readthedocs.io/en/v0.5.0/050-breaking-changes.html#explicitness-requirements